### PR TITLE
added suggestion limit prop to AutocompleteInput

### PR DESCRIPTION
**Pull request by mohinderps**

### CODE REVIEW

**Comment by oksuz** [comment]
@djhi I need this improvement also.

File: packages/ra-ui-materialui/src/input/AutocompleteInput.js [hunk 1]
```diff
@@ -455,6 +462,7 @@ export class AutocompleteInput extends React.Component {
             options,
         } = this.props;
         const { suggestions, searchText } = this.state;
+        const slicedSuggestions = this.sliceSuggestions(suggestions);
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/input/AutocompleteInput.js]: The result will be a new array for each `render()`, which will force a rerender of the autosuggest even if the suggestions didn't change. This is bad, so you should slice when setting the suggestions in the state, not when reading the state.

File: packages/ra-ui-materialui/src/input/AutocompleteInput.js [hunk 2]
```diff
@@ -520,6 +528,7 @@ AutocompleteInput.propTypes = {
     translate: PropTypes.func.isRequired,
     translateChoice: PropTypes.bool.isRequired,
     inputValueMatcher: PropTypes.func,
+    suggestionLimit: PropTypes.number,
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/input/AutocompleteInput.js]: please document and unit test this new prop

**Comment by mohinderps** [comment]
Okay, Will do.

**Comment by fzaninotto** [comment]
Hi! Could you find a few minutes tu update your PR?

**Comment by fzaninotto** [comment]
No news for too long, closing.

**Pull request closed without merging**